### ISSUE TITLE
Problem building client image on RHEL4u4 (?)

### ISSUE THREAD

**Issue by dikim33** [oscar-cluster/oscar#262]
**Reported by bli on 18 Sep 2006 07:26 UTC**
Was testing the latest branch-5-0 on RHEL4u4 and ran into the following issue:

```
=============================================================================
== Running step 4 of the OSCAR wizard: Build OSCAR client image
=============================================================================

--> Step 4: Identified distro of clients: rhel 4
--> Step 4: distro repo: /tftpboot/distro/redhat-el-ws-4-i386
--> Step 4: oscar repo: /tftpboot/oscar/common-rpms,/tftpboot/oscar/rhel-4-i386
--> Step 4: Using RPM list: /opt/oscar-5.0a1r5347M/oscarsamples/rhel-4-i386.rpmlist
Executing command: mksiimage -A --name oscarimage --location /tftpboot/oscar/common-rpms,/tftpboot/oscar/rhel-4-i386,/tftpboot/distro/redhat-el-ws-4-i386 --filename /opt/oscar-5.0a1r5347M/oscarsamples/rhel-4-i386.rpmlist --arch i686 --path /var/lib/systemimager/images/oscarimage --filename=/tmp/oscar-install-rpmlist.23008
chroot: cannot run command `/sbin/ldconfig': No such file or directory
Image build finished.
```

Image creation subsequently failed.  It would be good if someone can confirm this...

**Comment by dikim33** [comment]
**Comment by bli on 19 Sep 2006 06:09 UTC**
Additional info with `OSCAR_VERBOSE=2`:

```
error: can't create transaction lock on /var/lib/systemimager/images/oscarimage/var/lock/rpm/transaction
```

I wonder if we have multiple instances of yum/rpm running?

**Comment by dikim33** [comment]
**Comment by bli on 19 Sep 2006 18:27 UTC**
According to alea from #oscar-cluster, he mentioned that new with RHEL4u4, the directory `/var/lock/rpm` needs to be present in the image before you can install RPMs.

**Comment by dikim33** [comment]
**Comment by bli on 19 Sep 2006 18:49 UTC**
Additional notes:

https://bugzilla.redhat.com/bugzilla/show_bug.cgi?id=151255

Looks like SystemInstaller needs to ensure `/var/lock/rpm` in the chroot is created prior to yum/rpm invocations?

I'm assigning to Erich for now, hopefully he can take a quick look at this...

**Comment by dikim33** [comment]
**Comment by efocht on 28 Sep 2006 14:40 UTC**
(In [5377]) Merged commit r5375 by efocht from 2006-09-27 18:51:31 +0200 (Wed, 27 Sep 2006)
with svnmerge_1by1 version v1.1 31.7.2006
from repository https://svn.oscar.openclustergroup.org/svn/oscar/trunk
to repository https://svn.oscar.openclustergroup.org/svn/oscar/branches/branch-5-0
Original commit message was:

Updated yume to version 2.4-1. Should fix issues with RHEL4u4 when installing
into empty images (bug #262). Also contributes indirectly to fixing bug #33
(adapted yume to work better from clients).

Closes #262
